helix-term/src/commands/lsp.rs

```diff
@@ -957,6 +957,14 @@ pub fn signature_help_impl(cx: &mut Context, invoked: SignatureHelpInvoked) {
                 return;
             }
 
+            // If the signature help invocation is automatic, don't show it outside of Insert Mode:
+            // it very probably means the server was a little slow to respond and the user has
+            // already moved on to something else, making a signature help popup will just be an
+            // annoyance, see https://github.com/helix-editor/helix/issues/3112
+            if !was_manually_invoked && editor.mode != Mode::Insert {
```

Comment:
I think even if it's manually invoked, the same thing could happen right? Should we remove the first condition?

Reply:
Yes it can but that's expected, there is "space+k" to get documentation, that command only works outside of insert mode.